services/librariesio/librariesio-dependencies.service.js

```diff
@@ -1,134 +1,137 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const { checkErrorResponse } = require('../../lib/error-helper')
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class LibrariesioDependencies extends LegacyService {
+function transform({ dependencies }) {
+  return {
+    deprecatedCount: dependencies.filter(dep => dep.deprecated).length,
+    outdatedCount: dependencies.filter(dep => dep.outdated).length,
+  }
+}
+
+function renderDependenciesBadge({ deprecatedCount, outdatedCount }) {
```

Comment:
I know that all these paths weren't fully tested in the legacy service implementation, but would it be worth adding a test/tests to validate this?

These two functions (`transform` and `renderDependenciesBadge`) are ones I've been testing with `sazerac` and friends lately, although it looks like testing with that approach may not be straightforward with this current implementation

Reply:
I could add some sazerac tests of the render function. I could add some of `transform` too, though I'm not sure they're as meaningful as it's so trivial.

Reply:
It's not a blocker for me, just noticed that the branch coverage for the librariesio family is only 66.67%

Reply:
Have added some tests.